### PR TITLE
Initial Routing

### PR DESCRIPTION
**Pull request by graywolf336**
- Router supports middleware
- On config load, added a map of the info/error short links

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -40,6 +43,16 @@ func (c *config) load(filePath string) error {
 		return err
 	}
 
+	c.Hardcoded.Errors = map[string]models.ShortLink{}
+	for _, e := range c.Hardcoded.Error {
+		c.Hardcoded.Errors[e.Key] = e
```
Comment (geekgonecrazy) [inline config/config.go:76]: What if key is missing or empty?

File: controllers/errors.go [hunk 2]
```diff
@@ -0,0 +1,18 @@
+package controllers
+
+import (
+	"net/http"
+
+	"github.com/FideTech/yaus/config"
+)
+
+//ErrorHandler handles the hard coded "/e/" short links
+func ErrorHandler(w http.ResponseWriter, r *http.Request, value string) {
```
Comment (geekgonecrazy) [inline controllers/errors.go]: For sanity I wonder if should call it ErrorLinkHandler since ErrorHandler sounds like handling errors

Comment (graywolf336) [reply controllers/errors.go]: Great point! I will change that

File: config/config.go (hunk 1 above)
Comment (graywolf336) [reply config/config.go:76]: And what if there are duplicates? For sure need to handle those cases

**Review by geekgonecrazy — APPROVED** [review]

**Pull request merged**